qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -152,12 +153,9 @@ def preferred_init_points(self):
             return np.zeros(self.reps * len(self.operators), dtype=float)
 
     def _build(self):
-        if self._data is not None:
+        if self._valid:
             return
 
-        # need to check configuration here to ensure the operators are not None
-        self._check_configuration()
```

Comment:
Removing this will not properly raise the error that the operators are empty (which I guess is why you had to change the error in the evolved op ansatz test). I think it would be better to check the configuration as soon as possible, before any computations are done. 

However, if you really do want to remove the `check_configuration` from here, I think we should move the check if `self.operators` is `None` in this function.

Reply:
Again, I see the point, so will test it out over the weekend.